streaming/base/shuffle/py1e.py

```diff
@@ -107,6 +116,12 @@ def get_shuffle_py1e(shard_sizes: NDArray[np.int64],
             # Update sample offset for the next shard.
             cn_sample_offset += span_size
 
+        # If warn_user is true, this means the block size for shifts was smaller than a span size.
+        # This will result in no shuffling being done on that span aka shard part, so warn user.
+        if warn_user:
+            warnings.warn('Shuffle block size was smaller than shard size for some shards. This \
```

Comment:
Is there a way to validate this behavior using a unit test?

Reply:
Oh yeah let me add that!

Reply:
added unit test! I didn't even know pytest could detect warnings...